pip_audit/format/cyclonedx.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+
+from cyclonedx import output  # type: ignore
+from cyclonedx.model.bom import Bom  # type: ignore
+from cyclonedx.model.component import Component  # type: ignore
+from cyclonedx.model.vulnerability import Vulnerability  # type: ignore
+from cyclonedx.parser import BaseParser  # type: ignore
+
+import pip_audit.service as service
+
+from .interface import VulnerabilityFormat
+
+
+class PipAuditResultParser(BaseParser):
+    def __init__(self, result: Dict[service.Dependency, List[service.VulnerabilityResult]]):
+        super().__init__()
+
+        for (dep, vulns) in result.items():
+            c = Component(name=dep.name, version=str(dep.version))
+
+            # TODO(ww): Figure out if/how we want to include other dependency
+            # metadata in the BOM, such as author, license, etc.
```

Comment:
Braindump: we probably want to keep the `Dependency` model itself as small as possible, so I'm thinking something like this:

```python
@dataclass(frozen=True)
class Dependency:
    name: str
    version: Version
    metadata: Metadata = Metadata()
```

...where `Metadata` is a bunch of `Optional[T]`s that dependency sources can use to opportunistically enhance the SBOM output.

Reply:
I like that approach. 👍

Reply:
Yeah, I'll turn these TODOs into issues where we can figure out a reasonable approach to this.

Reply:
https://github.com/trailofbits/pip-audit/issues/123